tests/components/gree/test_switch.py

```diff
@@ -24,11 +26,31 @@
 ENTITY_ID_XFAN = f"{DOMAIN}.fake_device_1_xfan"
 
 
-async def async_setup_gree(hass):
+async def async_setup_gree(hass, enable_health=True):
     """Set up the gree switch platform."""
-    MockConfigEntry(domain=GREE_DOMAIN).add_to_hass(hass)
-    await async_setup_component(hass, GREE_DOMAIN, {GREE_DOMAIN: {DOMAIN: {}}})
-    await hass.async_block_till_done()
+    entry = MockConfigEntry(domain=GREE_DOMAIN, data={GREE_DOMAIN: {DOMAIN: {}}})
+    entry.add_to_hass(hass)
+
+    if enable_health:
+        with patch(
+            "homeassistant.components.gree.switch.GreeHealthModeSwitchEntity.entity_registry_enabled_default",
+            True,
```

Comment:
I've spent a long time trying to enable an entity through entityRegistry and various other means, but finally settled for this.

Reply:
use `entity_registry_enabled_by_default` mocker as in
`async def async_setup_gree(hass, entity_registry_enabled_by_default, enable_health=True):`

It will then automatically enable any disabled entities for you

Reply:
Thanks for the tip, new commit coming up!